src/Raven.Server/Documents/Indexes/MapReduce/OutputToCollection/OutputReduceToCollectionCommand.cs

```diff
@@ -199,11 +199,30 @@ public void Dispose()
         }
     }
 
-    public class OutputReduceToCollectionCommand : TransactionOperationsMerger.MergedTransactionCommand
+    public abstract class OutputReduceAbstractCommand : TransactionOperationsMerger.MergedTransactionCommand
+    {
+        protected readonly DocumentDatabase _database;
+
+        protected OutputReduceAbstractCommand(DocumentDatabase database)
+        {
+            _database = database;
+        }
+
+        public void ArtificialDelete(DocumentsOperationContext context, string key)
+        {
+            _database.DocumentsStorage.Delete(context, key, DocumentFlags.Artificial | DocumentFlags.FromIndex);
```

Comment:
As I see that's the main change - now we do pass `DocumentFlags.Artificial | DocumentFlags.FromIndex` flags to delete.

Reply:
Can you elaborate more why is it needed?

Reply:
I've merged the PR so we can proceed with Prod deployment. Please address @arekpalinski concerns

Reply:
This is one of the fixes... at one place we weren't passing those flags also on a `put` of a document.
In general, you don't want the tombstone/document generated here to be replicated as well..

For tombstone, most of the time it worked.. because we create the tombstone flags, based on the document flags, so if document flags contained  `Artificial` the tombstone would as well.

Reply:
Does it mean that from now even if you modify an artificial document manually then the Artificial flag will be preserved?

Reply:
No, modifying is simply a `put` where we do not relay on existing flags... But if this document will be changed again by the index, then it will have those flags once more

Reply:
ok